contracts/adapters/HegicAdapter.sol

```diff
@@ -352,19 +352,55 @@ contract HegicAdapter is IProtocolAdapter {
     }
 
     /**
-     * @notice Function to bulk claim rHEGIC2 rewards from liquidity utilization
+     * @notice Function to get rHEGIC2 rewards claimable from liquidity utilization
+     * @param rewardsAddress is the address of the rewards contract (either for eth or wbtc)
+     * @param optionIDs is an array of all the option ids we want to claim rewards for
+     */
+    function rewardsClaimable(address rewardsAddress, uint256[] calldata optionIDs)
+        external
+        view
+        returns (uint256 rewardsAmount)
+    {
+        IHegicRewards rewardsContract = IHegicRewards(rewardsAddress);
+
+        uint256 i = 0;
+
+        while (i < optionIDs.length && optionIDs[i] > 0) {
+          (, , , uint _amount, , uint _premium, , ) = rewardsContract.hegicOptions().options(optionIDs[i]);
+          if(!rewardsContract.rewardedOptions(optionIDs[i])){
+            rewardsAmount = rewardsAmount.add(_amount.div(100).add(_premium)
+            .mul(rewardsContract.rewardsRate())
+            .div(1e8));
+          }
+          i += 1;
+        }
+    }
+
+    /**
+     * @notice Function to get rHEGIC2 rewards claimable from liquidity utilization
      * @param rewardsAddress is the address of the rewards contract (either for eth or wbtc)
      * @param optionIDs is an array of all the option ids we want to claim rewards for
      */
     function claimRewards(address rewardsAddress, uint256[] calldata optionIDs)
         external
+        returns (uint256 rewardsAmount)
     {
         IHegicRewards rewardsContract = IHegicRewards(rewardsAddress);
+
         uint256 i = 0;
-        while (i < optionIDs.length) {
-          rewardsContract.getReward(optionIDs[i]);
+
+        uint256 balanceBefore = rewardsContract.hegic().balanceOf(address(this));
+
+        while (i < optionIDs.length && optionIDs[i] > 0) {
+          try rewardsContract.getReward(optionIDs[i]) {} catch {}
           i += 1;
         }
+
+        uint256 balanceAfter = rewardsContract.hegic().balanceOf(address(this));
+
+        rewardsAmount = balanceAfter.sub(balanceBefore);
+        require(rewardsAmount > 0, "No rewards to claim");
+        rewardsContract.hegic().safeTransfer(msg.sender, rewardsAmount);
```

Comment:
Emit an event at the end of this function, e.g. `RewardClaimed`

Reply:
Needed for indexing with [The Graph](http://thegraph.com)

Reply:
did you miss the event emission [here](https://github.com/ribbon-finance/structured-products/blob/5a814fa0a272f6c6412436a93e0dfc47c957e0a3/contracts/instruments/RibbonVolatility.sol#L468)
or did you see it and want it moved to hegicadapter?

Reply:
Oops, missed it. Let's leave it in RibbonVol